src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs

```diff
@@ -547,6 +547,8 @@ private static Task GetHostEntryOrAddressesCoreAsync(string hostName, bool justR
             {
                 if (NameResolutionPal.SupportsGetAddrInfoAsync)
                 {
+#pragma warning disable CS0162 // Unreachable code detected -- SupportsGetAddrInfoAsync is a constant on *nix.
```

Comment:
Where is the warning restored?

Reply:
The async tests didn't exist before #34633, but I'd like to keep them as they test the Windows-PAL / everything where `NameResolutionPal.SupportsGetAddrInfoAsync = true`.

Reply:
What I meant was, where is the corresponding `#pragma warning restore CS0162`?  I don't see it, which would mean this warning is being disabled accidentally for the rest of the file.

Reply:
I wondered this myself, as it was in the `Dns`-class without a #pragma restore, so just "copyed" this way over.
Happy that you confirmed that a restore is necessary.

Done with 8de5e3245a4a0fc02dd52b36615ac89802e002c5